### ISSUE TITLE
add support for @throws annotation

### ISSUE THREAD

**Issue by JanTvrdik** [phpstan/phpdoc-parser#4]

**Comment by SignpostMarv** [comment]
I'm thinking that phpstan/phpstan or phpstan/phpstan-strict-rules should throw an error (with file:line reference to where relevant docblock line was defined) **if** `@throws` says it throws something it doesn't, **but not** if it's defined on an *interface*

i.e.
```php
class Foo {
    /**
    * @throws RuntimeException when something goes wrong
    */
    public static function Bar() : void { // should maybe trigger error?
    }
}

interface FooTwo {
    /**
    * @throws RuntimeException when something goes wrong
    */
    public static function Bar() : void; // by virtue of being an interface, should not trigger an error
}

class FooThree implements FooTwo {
    public static function Bar() : void { // should maybe trigger error, but perhaps only on phpstan/phpstan-strict-rules ?
    }
}
```

**Comment by ondrejmirtes** [comment]
Hi, the annotation is already supported: https://github.com/phpstan/phpdoc-parser/commit/f8a1d21be13d637a9ccee0aa44b573fe530bad84

Implementation in PHPStan is on the roadmap and will have to be quite complex.

**Comment by pura** [comment]
Exploring this amazing tool. Was curious if this has been implemented or is on the roadmap.

**Comment by pura** [comment]
Thank you @ondrejmirtes that looks like what I'm looking for.
